library/kotlin/test/io/envoyproxy/envoymobile/RequestBuilderTest.kt

```diff
@@ -0,0 +1,105 @@
+package io.envoyproxy.envoymobile
+
+import org.assertj.core.api.Assertions.assertThat
+import org.junit.Test
+import java.net.URL
+
+class RequestBuilderTest {
+
+  @Test
+  fun `adding request data should have body present in request`() {
```

Comment:
Probably worth talking about how we want to write our test names.

The one I chose was `[condition] should [expectation]`. I tend to like to avoid adding the method or prefix with `test` (more explanation on the alternatives)

Alternatives:
* `should [expectation] when [condition]`
* `[method name] should [expectation] when [condition]` -- this adds overhead onto identifying what the actual method is that we want to test
* `test [method name] should [expectation] when [condition]` -- I personally like this the least because I feel having a prefix of `test` is redundant

Reply:
I generally agree with the sentiment of `[condition] should [expectation]`:
- iOS tests are identified by the `test` prefix, so we **must** have these prefixes in Swift/ObjC
- `[method name] should [expectation] when [condition]` - I think this one depends a bit on the test. In some cases I could see it being valuable to clarify

Reply:
The other thing which I am doing here is leveraging Kotlin's feature of declaring methods as sentence strings -- not sure if you have opinions here @junr03 @rebello95 @goaway

Reply:
Can't do the sentence strings in Swift, but not opposed to using it in Kotlin

Reply:
> [method name] should [expectation] when [condition] - I think this one depends a bit on the test. In some cases I could see it being valuable to clarify

Yeah, there might be exceptions but for the most part the cases that I've run across where the method name is present are when the method actually describes the behavior (like `track event should publish to the pipeline`).

Reply:
My take:

- I think this looks good. I think it works especially well when your unit under test has a small scope with limited behavior (also a characteristic of good design/architecture).
- I think we should still be ok using `[method name] should [expectation] (when [condition])` for more complex cases where it would add clarity. I do think we should be consistent in format for a given file/unit.
- I think the "sentence" function names in kotlin are nice and see no reason not to use them.

Reply:
> I think we should still be ok using [method name] should [expectation] (when [condition]) for more complex cases where it would add clarity. I do think we should be consistent in format for a given file/unit.

I agree!